pkg/trace/api/responses.go

```diff
@@ -90,18 +91,32 @@ func (wc *writeCounter) N() uint64 { return atomic.LoadUint64(&wc.n) }
 // httpRateByService outputs, as a JSON, the recommended sampling rates for all services.
 // It returns the number of bytes written and a boolean specifying whether the write was
 // successful.
-func httpRateByService(w http.ResponseWriter, dynConf *sampler.DynamicConfig) (n uint64, ok bool) {
+func httpRateByService(ratesVersion string, w http.ResponseWriter, dynConf *sampler.DynamicConfig) (n uint64, ok bool) {
+	wc := newWriteCounter(w)
+	var err error
+	defer func() {
+		n, ok = wc.N(), err == nil
+		if err != nil {
+			tags := []string{"error:response-error"}
+			metrics.Count(receiverErrorKey, 1, tags, 1)
+		}
+	}()
+
+	currentState := dynConf.RateByService.GetNewState(ratesVersion) // this is thread-safe
 	w.Header().Set("Content-Type", "application/json")
+	w.Header().Set(headerRatesPayloadVersion, currentState.Version)
+
+	if ratesVersion != "" && ratesVersion == currentState.Version {
+		// rates weren't changed
+		_, err = wc.Write([]byte("{}"))
```

Comment:
👍🏻 to writing `{}`

If we think there are any risks with this being an empty object, we could always do `{"rate_by_service": {}}` to include the existing key.

Reply:
Hi @brettlangdon . Now, we respond `{}` only if `Datadog-Rates-Payload-Version` exists. Which means, if the tracer is older, we never return `{}`.